modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -477,7 +481,11 @@ private void updateTableSkin() {
         }
     }
 
-    TableViewSkinBase<?,?,?,?,?> getTableSkin() {
+    /**
+     *
+     * @return The {@link TableViewSkinBase} in which this TableColumnHeader is inserted.
+     */
+    protected TableViewSkinBase<?,?,?,?,?> getTableSkin() {
```

Comment:
IMHO, this method is only used when interacted with TableSkinUtils. Since the TableSkin is created way before this class, it is not compulsory to open this method. But I believe that if the original TableCOlumnHeader is using it so much, anyone extending the TableColumnHeader will also need a convenient access to the TableSkin.

Reply:
definitely needed in custom column headers, mainly to configure itself based on the table/skin (been there and suffered ;) There's a catch, though: in the "early" stages of the header's life-cycle - that is after being created and before the header row has been set - the skin is null. Plus there is no clean (nor any dirty) way to know when exactly the skin is available. Historically, its safe (read: not null) access got lost in a bug fix in NestedTableColumnHeader which also removed the skin as parameter in the constructor.